ansible/roles/aws_service_limit/library/aws_service_limit.py

```diff
@@ -0,0 +1,187 @@
+    Collects and arranges Trusted Advisor service limits
+
+    Param: support connection object
+    Returns: list of service limits; each service limit is a dict containing:
+        - region (string)
+        - service (string)
+        - limit_name (string)
+        - limit_value (integer)
+        - current_value (integer)
+        - status (string)
+    '''
+    # look up the service limits ID
+    ta_query = tr_adv.describe_trusted_advisor_checks(language='en')
+    service_limits_id = None
+    for check in ta_query['checks']:
+        if check['name'] == "Service Limits":
+            service_limits_id = check['id']
+
+    # Gather results from Trusted Advisor
+    ta_results = tr_adv.describe_trusted_advisor_check_result(checkId=service_limits_id)
+
+    # extract limits from the most recent TA report
+    limits = []
+    for res in ta_results['result']['flaggedResources']:
+        limits.append({'region' : res['metadata'][0],
+                       'service' : res['metadata'][1],
+                       'limit_name' : res['metadata'][2],
+                       'limit_value' : res['metadata'][3],
+                       'current_value' : res['metadata'][4],
+                       'status' : res['metadata'][5]})
+    return limits
+
+def main():
+    """
+    get service limits
+    :return:
+    """
+
+    result = {}
+
+    # Including ec2 argument spec
+    module = AnsibleModule(argument_spec=ec2_argument_spec(),
+                           supports_check_mode=True)
+
+    # Set up connection
+    region, ec2_url, aws_connect_params = get_aws_connection_info(module,
+                                                                  boto3=HAS_BOTO3)
+
+    # Set up connection
+    if region:
+        try:
+            ec2_conn = boto3_conn(module,
+                                  conn_type='client',
+                                  resource='ec2',
+                                  region=region,
+                                  endpoint=ec2_url,
+                                  **aws_connect_params)
+
+            ta_conn = boto3_conn(module,
+                                 conn_type='client',
+                                 resource='support',
+                                 region=region,
+                                 endpoint=ec2_url,
+                                 **aws_connect_params)
+        except (botocore.exceptions.NoCredentialsError,
+                botocore.exceptions.ProfileNotFound) as exc:
+            module.fail_json(msg=exc.message,
+                             exception=traceback.format_exc(),
+                             **camel_dict_to_snake_dict(exc.response))
+    else:
+        module.fail_json(msg="AWS region must be specified (like: us-east-1)")
+
+    result['aws_account_attributes'] = account_attributes(ec2_conn)
+    result['aws_service_limits'] = service_limits(ta_conn)
+
+    # Send exit
+    module.exit_json(msg="Retrieved ec2 account attributes.", ansible_facts=result)
```

Comment:
does returning through 'ansible_facts' mean the returned results will be dumped into Ansible's facts for the specific host? if so, should we be associating this non-host-specific data with a host?

our previous attempts at returning data in our modules simply exit_json(results=<result_object_here>). you can then just register the results of the task and access it as a play/playbook variable.